### PR TITLE
Add hip related functions

### PR DESCRIPTION
**Pull request by rbramand-xilinx**
#### Problem solved by the commit
Added new hip module function as well as common utility function for creating host buffer

#### Risks (if any) associated the changes in the commit
Low

#### What has been tested and how, request additional testing if necessary
Tested build flow on Linux and Windows
Ran xrt_flow test on Linux and it passes

#### Documentation impact (if any)
NA

### CODE REVIEW

**Comment by rbramand-xilinx** [comment]
This PR contains changes from other PR (https://github.com/Xilinx/XRT/pull/8110) which is open so added do not merge tag.
Once that PR gets merged I will rebase this and remove the tag.

File: src/runtime_src/hip/core/module.cpp
Comment (stsoe) [inline src/runtime_src/hip/core/module.cpp:34]: Why this const_cast?  xclbin ctor takes const axlf*

**Review by stsoe — CHANGES_REQUESTED** [review]
I don't think we should copy xrt_flow into the xrt repo.  This kind of test case belongs under testcases., or should be much simplified.     Copying a boatload of code means two places to maintain same boatload of code, not ideal.  We did the same mistakenly in XRT-MCDM for an even simpler dputest, and today that test doesn't work and no one has the energy to check and fix it.

**Comment by rbramand-xilinx** [comment]
Hi @stsoe, updated the PR to remove xrt_flow test and I will push this test to testcases repo.
Updated the PR description appropriately. Please review

File: tests/hip/common/common.h
Comment (stsoe) [inline tests/hip/common/common.h:127]: Not needed in this, doesn't hurt either.
```suggestion
  hip_test_host_bo(size_t size, unsigned int flags) : _buffer(nullptr) {
```

File: tests/hip/common/common.h
Comment (stsoe) [inline tests/hip/common/common.h:133]: Dtors are automatically noexcept.  The program will crash if `test_hip_check` throws.  If the dtor executes any code that can throw, then it would be better to try/catch and attempt to exit grace fully.
```suggestion
  ~hip_test_host_bo()  {
    test_hip_check(hipHostFree(_buffer));
  }
```

**Review by stsoe — APPROVED** [review]
Looks good.  Minor comments.

**Review by stsoe — APPROVED** [review]

**Review by chvamshi-xilinx — APPROVED** [review]

**Pull request merged**
